src/docker.rs

```diff
@@ -388,16 +388,19 @@ impl DockerClientCertResolver {
                 path: self.ssl_key.to_owned(),
             });
         };
-
-        let signing_key = any_supported_type(&key).map_err(|_| CertParseError {
-            path: self.ssl_key.to_owned(),
-        })?;
+        let signing_key = CryptoProvider::get_default()
+            .expect("no process-level CryptoProvider available -- call CryptoProvider::install_default() before this point")
```

Comment:
We probably shouldn't panic here, as we would need to document clearly that it can panic in the connection methods. Could you create a new error type in `errors.rs` with the appropriate message and return that ?

Reply:
The underlying reasoning was if the user failed to install a provider, this error is just unrecoverable and the code should just panic. 
That's something that crates inside the `rustls` organization do.
For instance, see those lines in [rustls](https://github.com/rustls/rustls/blob/e52853a08fa366bb6f3253e932f5608c8b580d39/rustls/src/crypto/mod.rs#L265) and [rustls-platform-verifier](https://github.com/rustls/rustls-platform-verifier/blob/1ba897603508c55a7c350266238b9e9f2c7a578a/rustls-platform-verifier/src/verification/mod.rs#L113)

Anyway I don't feel strongly about this and would be happy to change if you prefer the other way.

Reply:
Ok. It's understandable perhaps to panic with incorrect compile time flags, but we don't quite know how a docker client is invoked (maybe in a parallel thread), would be best to return an error in any case.